hbase-server/src/main/java/org/apache/hadoop/hbase/master/cleaner/SnapshotCleanerChore.java

```diff
@@ -0,0 +1,99 @@
+import org.apache.hadoop.hbase.util.EnvironmentEdgeManager;
+import org.apache.hbase.thirdparty.org.apache.commons.collections4.CollectionUtils;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * This chore, every time it runs, will try to delete snapshots that are expired based on TTL in
+ * seconds configured for each Snapshot
+ */
+@InterfaceAudience.Private
+public class SnapshotCleanerChore extends ScheduledChore {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotCleanerChore.class);
+  private static final String SNAPSHOT_CLEANER_CHORE_NAME = "SnapshotCleaner";
+  private static final String SNAPSHOT_CLEANER_INTERVAL = "hbase.master.cleaner.snapshot.interval";
+  private static final int SNAPSHOT_CLEANER_DEFAULT_INTERVAL = 600 * 1000; // Default 10 min
+  private static final String DELETE_SNAPSHOT_EVENT =
+          "Eligible Snapshot for cleanup due to expired TTL.";
+
+  private final SnapshotManager snapshotManager;
+
+  public SnapshotCleanerChore(Stoppable stopper, Configuration configuration,
+          SnapshotManager snapshotManager) {
+    super(SNAPSHOT_CLEANER_CHORE_NAME, stopper, configuration.getInt(SNAPSHOT_CLEANER_INTERVAL,
+            SNAPSHOT_CLEANER_DEFAULT_INTERVAL));
+    this.snapshotManager = snapshotManager;
+  }
+
+  @Override
+  protected void chore() {
+    LOGGER.trace("Snapshot Cleaner Chore is starting up...");
+    try {
+      List<SnapshotProtos.SnapshotDescription> completedSnapshotsList =
+              this.snapshotManager.getCompletedSnapshots();
+      if (CollectionUtils.isNotEmpty(completedSnapshotsList)) {
+        for (SnapshotProtos.SnapshotDescription snapshotDescription : completedSnapshotsList) {
+          long snapshotCreatedTime = snapshotDescription.getCreationTime();
+          long snapshotTtl = snapshotDescription.getTtl();
+          /*
+           * Backward compatibility after the patch deployment on HMaster
+           * Any snapshot with negative or zero ttl should not be deleted
+           * Default ttl value specified by {@HConstants.DEFAULT_SNAPSHOT_TTL}
+           */
+          if (snapshotCreatedTime > 0 && snapshotTtl > 0 &&
+                  snapshotTtl < (Long.MAX_VALUE / 1000)) {
+            long currentTime = EnvironmentEdgeManager.currentTime();
+            if ((snapshotCreatedTime + snapshotTtl * 1000) < currentTime) {
+              LOGGER.info("Event: {} Name: {}, CreatedTime: {}, TTL: {}, currentTime: {}",
+                      DELETE_SNAPSHOT_EVENT, snapshotDescription.getName(), snapshotCreatedTime,
+                      snapshotTtl, currentTime);
+              deleteExpiredSnapshot(snapshotDescription);
+            }
+          }
+        }
+      }
+    } catch (IOException e) {
+      LOGGER.error("Error while cleaning up Snapshots. Stopping Snapshot Cleaner Chore...", e);
```

Comment:
"Stopping Snapshot Cleaner Chore..." might be confusing since it's still scheduled next time.